### PR TITLE
Detect type to decode

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuck@heptio.com>

### CODE REVIEW

File: pkg/results/reader.go [hunk 1]
```diff
@@ -155,8 +156,20 @@ func ExtractIntoStruct(predicate func(string) bool, path string, info os.FileInf
 		if !ok {
 			return errors.New("info.Sys() is not a reader")
 		}
+		// TODO(chuckha) there must be a better way
```
Comment (timothysc) [inline pkg/results/reader.go]: upstream has the "universal decoder", I'll have to dig up a link.

Comment (timothysc) [reply pkg/results/reader.go]: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/util.go#L244 

legacyscheme.Codecs.UniversalDecoder()

Comment (chuckha) [reply pkg/results/reader.go]: i looked at it briefly but it looked like it was pretty k8s specific. Maybe worth looking at deeper though.

For now, I went with the simplest approach possible.

Comment (ncdc) [reply pkg/results/reader.go]: UniversalDecoder only decodes what kube's apimachinery can handle. That's usually json, yaml, and proto.

**Pull request merged**